config/fields/object.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+use Kirby\Data\Data;
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Form\Form;
+use Kirby\Toolkit\A;
+
+return [
+	'props' => [
+		/**
+		 * Unset inherited props
+		 */
+		'after'       => null,
+		'before'      => null,
+		'autofocus'   => null,
+		'icon'        => null,
+		'placeholder' => null,
+
+		/**
+		 * Set the default values for the object
+		 */
+		'default' => function (array $default = []) {
+			return $default;
+		},
+
+		/**
+		 * Fields setup for the object form. Works just like fields in regular forms.
+		 */
+		'fields' => function (array $fields = []) {
+			return $fields;
+		}
+	],
+	'computed' => [
+		'default' => function () {
+			return $this->form($this->default)->values();
+		},
+		'fields' => function () {
+			if (empty($this->fields) === true) {
+				throw new Exception('Please provide some fields for the object');
+			}
+
+			return $this->form()->fields()->toArray();
+		},
+		'value' => function () {
+			$data = Data::decode($this->value, 'yaml');
+
+			return $this->form($data)->values();
+		}
+	],
+	'methods' => [
+		'form' => function (array $values = []) {
+			return new Form([
+				'fields' => $this->attrs['fields'],
+				'values' => $values,
+				'model'  => $this->model
+			]);
+		},
+	],
+	'save' => function ($value) {
+		return $this->form($value)->content();
+	},
+	'validations' => [
+		'object' => function ($value) {
+			$errors = $this->form($value)->errors();
+
+			if (empty($errors) === false) {
+				// use the first error for details
+				$error = A::first($errors);
+
+				throw new InvalidArgumentException([
+					'key'  => 'object.validation',
+					'data' => [
+						'label' => $error['label'],
+						'message' => A::first($error['message'])
```

Comment:
Why are we here only using the first error and the first message?

Reply:
Agreed. Should be validate all fields and shows all errors.

This is also related with https://github.com/getkirby/kirby/pull/4629 enhancement since there are multiple field errors. So error should be thrown like following:

````php
throw new InvalidArgumentException(implode("\n", ['Error one', 'Error two', 'Error three']));
````